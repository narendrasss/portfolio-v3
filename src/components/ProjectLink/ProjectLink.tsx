import React from 'react';
import Link from '../base/Link';
import { Flex } from 'rebass';
import Heading from '../base/Heading';
import Text from '../base/Text';
import Box from '../base/Box';
import List from '../base/List';

const ProjectLink: React.FC<PageFrontmatter> = ({
  path,
  title,
  description,
  tech,
}) => (
  <Link to={path}>
    <Flex alignItems="center">
      <Heading fontSize={1}>{title}</Heading>
      <Box>
        <Text fontSize="2rem">{description}</Text>
        <Flex>
          <List items={tech}>
            {technology => (
              <Text key={technology} fontWeight={300} fontSize="2rem">
                {technology}
              </Text>
            )}
          </List>
        </Flex>
      </Box>
    </Flex>
  </Link>
);

export default ProjectLink;
