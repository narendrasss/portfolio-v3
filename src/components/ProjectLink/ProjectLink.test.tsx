import React from 'react';
import { render } from 'react-testing-library';
import ProjectLink from './ProjectLink';

describe('ProjectLink', () => {
  const mockFrontmatter = {
    path: '/projects/coin',
    title: 'Coin',
    tech: ['react', 'typescript'],
    description: 'Web based financial management',
  };

  it('renders without crashing', () => {
    const { asFragment } = render(<ProjectLink {...mockFrontmatter} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the project title', () => {
    const { queryByText } = render(<ProjectLink {...mockFrontmatter} />);
    expect(queryByText(mockFrontmatter.title)).toBeTruthy();
  });

  it('renders the project tech stack', () => {
    const { queryByText } = render(<ProjectLink {...mockFrontmatter} />);
    mockFrontmatter.tech.forEach(tech => {
      expect(queryByText(tech)).toBeTruthy();
    });
  });

  it('renders the project description', () => {
    const { queryByText } = render(<ProjectLink {...mockFrontmatter} />);
    expect(queryByText(mockFrontmatter.description)).toBeTruthy();
  });
});
