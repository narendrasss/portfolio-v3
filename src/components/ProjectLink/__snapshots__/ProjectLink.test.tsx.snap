// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLink renders without crashing 1`] = `
<DocumentFragment>
  <a
    class="sc-bdVaJa sc-htoDjs jssEWI sc-htpNat dRQZBK"
    color="black"
    font-family="sans"
    font-size="0,1"
    href="/projects/coin"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH pjOvZ"
    >
      <h1
        class="sc-bdVaJa sc-htpNat sc-bxivhb gBkMPw"
        font-size="2"
        font-weight="bold"
      >
        Coin
      </h1>
      <div
        class="sc-bdVaJa sc-dnqmqq bzPRsb"
      >
        <p
          class="sc-bdVaJa sc-htpNat ejPUVP"
          font-family="sans"
          font-size="0,1"
        >
          Web based financial management
        </p>
        <div
          class="sc-bdVaJa sc-bwzfXH pjOvZ"
        >
          <p
            class="sc-bdVaJa sc-htpNat ejPUVP"
            font-family="sans"
            font-size="0,1"
          >
            react
          </p>
          <p
            class="sc-bdVaJa sc-htpNat ejPUVP"
            font-family="sans"
            font-size="0,1"
          >
            typescript
          </p>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
